### PR TITLE
Multi-contract layout

### PR DESCRIPTION
**Pull request by sanderpick**
@dtbuchholz and I collaborated on this today. The goal was to reduce the contract size and also give us some more flex on future things. The contract size on `main` is very close to the limit... meaning if we were to add _any_ other logic, we'd have to somehow break implementation code out, which is much harder retroactively. Some inputs to the approach:
- The main issue w/ contract size is the implementation code, not really related to storage
- But keeping the storage in the rigs contract feel limiting to future things. If it was elsewhere, we could always migrate data (can't migrate rigs contract data).
- Using a linked library to break off implementation and/or storage is a common approach, but wouldn't work because the linked lib address wouldn't be upgradable. 
- The parent-child contract pattern is a good fit. Essentially, deploy an entirely new and independent _upgradable_ contract for pilots, and simply call it from Rigs. Rigs stored the address of this contract, which we could change at any time. Pilots is _only callable from Rigs_.
- This gives us freedom on implementation and the ability to migrate pilot storage if we needed to.

This doesn't touch any train / pilot / park logic or storage layout, just code organization. We did add the ability to train and park in bulk. Bulk piloting was already possible. It's deployed here:
- Rigs: https://mumbai.polygonscan.com/address/0xa04ac9Ca2FE3Db8f372B1364CdaD6fc58453Fde8
- Pilots: https://mumbai.polygonscan.com/address/0xEAbC6eF167a61bA20bb77030bd2cf7eAa218f11D
- Garage app: https://garage.tableland.xyz/
- Animation: https://testnets.opensea.io/assets/mumbai/0xa04ac9ca2fe3db8f372b1364cdad6fc58453fde8/2

This also includes some minor updates to the garage app:
- Updates Rigs function signatures since train / park are now overloaded
- Updates the train / park modals with more text description of what's going on.

### CODE REVIEW

**Review by dtbuchholz — APPROVED** [review]
lgtm 🚀

**Review by datadanne — COMMENTED** [review]
Nice!

File: ethereum/contracts/ITablelandRigs.sol [hunk 1]
```diff
@@ -202,61 +159,62 @@ interface ITablelandRigs {
     function setRoyaltyReceiver(address receiver) external;
 
     /**
-     * @dev Pauses minting.
+     * @dev Initializes Rig pilots by creating the pilot sessions table.
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply ethereum/contracts/ITablelandRigs.sol:162]: ah yep, good catch. doc string needs an update. may we well change the function name while we're at it to `setPilots` which sounds more correct. mind doing that @dtbuchholz (no rush)?

File: ethereum/contracts/TablelandRigs.sol [hunk 2]
```diff
@@ -373,420 +336,158 @@ contract TablelandRigs is
...
-    function _createPilotSessionsTable() internal {
-        _pilotSessionsTableId = TablelandDeployments.get().createTable(
-            address(this),
-            SQLHelpers.toCreateFromSchema(
-                "id integer primary key,rig_id integer not null,owner text not null,pilot_contract text,pilot_id integer,start_time integer not null,end_time integer",
-                PILOT_SESSIONS_PREFIX
-            )
-        );
+    function initPilots(address pilotsAddress) external onlyOwner {
+        _pilots = ITablelandRigPilots(pilotsAddress);
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply ethereum/contracts/TablelandRigs.sol:358]: I hear that... part of the reason for doing this upgradable separate contract approach for pilots was to retain as much flex as possible for future unknown unknown things related to the TBL protocol. We can def lock this down (and ever renounce ownership of Rigs and Pilots) in the future. I don't think this adds any additional security concern on top of the Rigs contract already being upgradable.

File: ethereum/contracts/TablelandRigs.sol [hunk 3]
```diff
@@ -373,420 +336,158 @@ contract TablelandRigs is
...
-            pilotRig(tokenIds[i], pilotContracts[i], pilotIds[i]);
+            pilotRig(tokenIds[i], pilotAddrs[i], pilotIds[i]);
         }
     }
 
     /**
      * @dev See {ITablelandRigs-parkRig}.
      */
-    function parkRig(uint256 tokenId) public {
+    function parkRig(uint256 tokenId) public whenNotPaused {
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply ethereum/contracts/TablelandRigs.sol:458]: Related to above, we'd need this in place if we decided that the pilots data needs to be migrated (you'd have to pause to as to get a snapshot).

File: ethereum/contracts/TablelandRigPilots.sol [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+        return _parent;
+    }
+
+    /**
+     * @dev See {ITablelandRigPilots-pilotSessionsTable}.
+     */
+    function pilotSessionsTable()
+        external
+        view
+        onlyParent
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply ethereum/contracts/TablelandRigPilots.sol:114]: Yep, good call. PR approved!

**Pull request merged**